ui/lib/utils/telemetry.ts

```diff
@@ -8,15 +8,36 @@ export async function init() {
   // check option
   const res = await client.getInstance().getInfo()
   if (res?.data?.disable_telemetry === false) {
-    mixpanel.opt_in_tracking()
     // https://developer.mixpanel.com/docs/javascript-full-api-reference#mixpanelset_config
     mixpanel.set_config({
       batch_requests: true,
       persistence: 'localStorage',
     })
+    mixpanel.register({
+      // $initial_referrer and $initial_referring_domain overrides don't work in register()
+      // $initial_referrer: '',
+      // $initial_referring_domain: '',
+      $referrer: '',
+      $referring_domain: '',
+    })
+    mixpanel.opt_in_tracking()
   }
 }
 
-export function track(eventType: string, eventBody: object) {
-  mixpanel.track(eventType, eventBody)
+function getPathInUrlHash(): string {
+  const fullUrlHash = window.location.hash
+  const pos = fullUrlHash.indexOf('?')
+  if (pos === -1) {
+    return fullUrlHash
+  }
+  return fullUrlHash.substring(0, pos)
+}
+
+export function track(eventType: string, eventBody: object = {}) {
+  mixpanel.track(eventType, {
+    ...eventBody,
+    $current_url: getPathInUrlHash(),
```

Comment:
I prefer to override `current_url` in route events, otherwise other mixpanel features eithere cannot be used (due to incorrect current_url) or need similar pattern again.

Reply:
You mean, use mixpanel.register() before mixpanel.track(), right? Let me try it.